File: src/main/java/de/hdawg/rankinginfo/domain/Player.java
package de.hdawg.rankinginfo.domain;

import java.util.List;

public record Player(
    int dtbId,
    String lastname,
    String firstname,
    String nationality,
    String club,
    String federation,
    List<Club> clubs) {

  public Player {
    clubs = List.copyOf(clubs);
  }
}

File: src/main/kotlin/de/hdawg/rankinginfo/domain/Player.kt
package de.hdawg.rankinginfo.domain

/**
 * Represents a tennis player with their current club affiliation and full club history.
 *
 * @property dtbId unique DTB identifier encoding gender and year of birth
 * @property clubs all clubs the player has appeared under across all available ranking periods
 */
data class Player(
    val dtbId: Int,
    val lastname: String,
    val firstname: String,
    val nationality: String,
    val club: String,
    val federation: String,
    val clubs: List<Club> = emptyList(),
)
